Modules/ITS/src/ITSFhrTask.cxx

```diff
@@ -562,24 +564,36 @@ void ITSFhrTask::monitorData(o2::framework::ProcessingContext& ctx)
         if ((double)GBTLinkInfo->statistics.nTriggers >= 1e6 && (double)GBTLinkInfo->statistics.nTriggers < 1e6 + 10000) {
           mNoisyPixelNumber[mLayer][istave] = 0;
         }
-
         for (int ichip = 0 + (ilink * 3); ichip < (ilink * 3) + 3; ichip++) {
           std::unordered_map<unsigned int, int>::iterator iter;
+
+          if (mDoHitmapFilter == 1 && mTFCount > mCutTFForSparse) {
+            for (auto iter = mHitPixelID_InStave[istave][0][ichip].begin(); iter != mHitPixelID_InStave[istave][0][ichip].end();) {
+              if ((double)iter->second / GBTLinkInfo->statistics.nTriggers < 1.7e-3) { // 40 hits/cm^2 * 5 pixels/hits * 4.5 cm^2 / 1024 / 512 = 1.7e-3/pixel/event for physics
+                mHitPixelID_InStave[istave][0][ichip].erase(iter++);
+              } else
+                ++iter;
+            }
+          }
+
           for (iter = mHitPixelID_InStave[istave][0][ichip].begin(); iter != mHitPixelID_InStave[istave][0][ichip].end(); iter++) {
             if ((iter->second > mHitCutForNoisyPixel) &&
                 (iter->second / (double)GBTLinkInfo->statistics.nTriggers) > mOccupancyCutForNoisyPixel &&
                 ((double)GBTLinkInfo->statistics.nTriggers >= 1e6 && (double)GBTLinkInfo->statistics.nTriggers < 1e6 + 10000)) {
               mNoisyPixelNumber[mLayer][istave]++; // count only in 10000 events as soon as nTriggers is 1e6
```

Comment:
Since now you are removing the non-noisy pixels, we can count the noisy pixels w/o waiting for 1M events now (at least when filtering is applied) so to have an idea how many they are (i.e. how many survive the cut) from the TH2Poly.